src/differential/mod.rs

```diff
@@ -1,4 +1,4 @@
-use hashbrown::HashMap;
+use fnv::FnvHashMap as HashMap;
```

Comment:
Is there a reason you're rebinding `FnvHashMap` as `HashMap` in this file but not the others? I'd prefer to make the binding consistent across files unless there's a good reason not to.